### PR TITLE
fix #270

### LINKED ISSUE #270: xgi.load_xgi_data(dataset=None) error
`xgi.load_xgi_data(dataset=None)` lists available datasets as expected but then raises an error:

```python
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In [2], line 1
----> 1 xgi.load_xgi_data(dataset=None)

File ~/Dropbox (ISI Foundation)/WORK/SCIENCE/xgi/xgi/readwrite/xgi_data.py:73, in load_xgi_data(dataset, cache, read, path, nodetype, edgetype, max_order)
     70 else:
     71     data = _request_from_xgi_data(dataset)
---> 73 return convert.dict_to_hypergraph(
     74     data, nodetype=nodetype, edgetype=edgetype, max_order=max_order
     75 )

File ~/Dropbox (ISI Foundation)/WORK/SCIENCE/xgi/xgi/convert.py:649, in dict_to_hypergraph(data, nodetype, edgetype, max_order)
    647 H = empty_hypergraph()
    648 try:
--> 649     H._hypergraph.update(data["hypergraph-data"])
    650 except KeyError:
    651     raise XGIError("Failed to get hypergraph data attributes.")

TypeError: 'NoneType' object is not subscriptable
```

### PR DESCRIPTION
**Pull request by nwlandry**
Fixes #270 as well as other issues.

### CODE REVIEW

File: xgi/readwrite/xgi_data.py [hunk 1]
```diff
@@ -70,6 +77,7 @@ def load_xgi_data(
     else:
         data = _request_from_xgi_data(dataset)
 
+    # if users actually requested data and didn't just
```
Comment (maximelucas) [inline xgi/readwrite/xgi_data.py:80]: comment sentence unfinished?

Comment (nwlandry) [reply xgi/readwrite/xgi_data.py:80]: Good catch. I forgot to remove this!

File: xgi/readwrite/xgi_data.py [hunk 2]
```diff
@@ -113,37 +120,23 @@ def _request_from_xgi_data(dataset=None):
...
-    if r.ok:
-        index_data = r.json()
-
-        if dataset is None:
-            print("Available datasets are the following:")
-            print(*index_data, sep="\n")
-            return
-    else:
-        raise XGIError(f"Error: HTTP response {r.status_code}")
+    index_data = _request_json_from_url(index_url)
```
Comment (maximelucas) [inline xgi/readwrite/xgi_data.py:131]: this line does not print the available datasets if dataset is None anymore, right?
is it printed in lines 134-137 instead?

Comment (nwlandry) [reply xgi/readwrite/xgi_data.py:132]: Correct. The list of available datasets is printed in lines 57-62 and I added a comment to reflect that. It got too messy when I tried to keep in in the `_request_json_from_url` function.

File: xgi/readwrite/xgi_data.py [hunk 3]
```diff
@@ -113,37 +120,23 @@ def _request_from_xgi_data(dataset=None):
...
         print("Valid dataset names:")
         print(*index_data, sep="\n")
         raise XGIError("Must choose a valid dataset name!")
 
-    r = requests.get(index_data[key]["url"])
-    if r.ok:
-        return r.json()
-    else:
-        raise XGIError(f"Error: HTTP response {r.status_code}")
+    return _request_json_from_url(index_data[key]["url"])
```
[the remark this answers is not in the document]
Comment (nwlandry) [reply xgi/readwrite/xgi_data.py:140]: Correct. I made a helper function since I used it in 3 different places.

**Review by maximelucas — APPROVED** [review]
Good to go, thanks!

**Pull request merged**